pkg/types/clusterconfig/cluster_config.go

```diff
@@ -917,15 +934,21 @@ func (cc *Config) Validate(awsClient *aws.Client) error {
 	cc.AccountID = accountID
 
 	if cc.Bucket == "" {
-		bucketID := hash.String(accountID + cc.Region)[:8] // this is to "guarantee" a globally unique name
-		cc.Bucket = cc.ClusterName + "-" + bucketID
-	} else {
+		cc.Bucket = BucketName(accountID, cc.ClusterName, cc.Region)
+		// check if the bucket already exists in a different region for some reason
 		bucketRegion, _ := aws.GetBucketRegion(cc.Bucket)
 		if bucketRegion != "" && bucketRegion != cc.Region { // if the bucket didn't exist, we will create it in the correct region, so there is no error
 			return ErrorS3RegionDiffersFromCluster(cc.Bucket, bucketRegion, cc.Region)
```

Comment:
This error message can be:

`fmt.Sprintf("the %s bucket already exists but is in %s (your cluster is in %s); either change the region of your cluster to %s or delete your bucket to allow cortex to create the bucket for you in %s", bucketName, bucketRegion, clusterRegion, bucketRegion, clusterRegion),`